### PR TITLE
Passing `DataLocale` by reference

### PR DESCRIPTION
**Pull request by robertbastian**
Before:
```rust
provider.load(&DataRequest {
  locale: DataLocale::from(locale),
  metadata: Default::default(),
};
```

Now:

```rust
provider.load(DataRequest {
  locale: &DataLocale::from(locale),
  metadata: Default::default(),
};
```

This allows reusing `DataLocales` for multiple requests.

### CODE REVIEW

**Comment by robertbastian** [comment]
If you agree with my proposal on #2188 you could also review the next step instead, #2227

File: provider/adapters/src/fallback/mod.rs [hunk 1]
```diff
@@ -266,13 +285,11 @@ impl<'a> LocaleFallbackerWithConfig<'a> {
     /// When first initialized, the locale is normalized according to the fallback algorithm.
     ///
     /// [`Locale`]: icu_locid::Locale
-    pub fn fallback_for<'b, T>(&'b self, mut ro: T) -> LocaleFallbackIterator<'a, 'b, T>
-    where
-        T: AsMut<DataLocale>,
-    {
-        self.normalize(ro.as_mut());
+    pub fn fallback_for<'b>(&'b self, req: DataRequest) -> LocaleFallbackIterator<'a, 'b> {
+        let mut locale = req.locale.clone();
```
Comment (sffc) [inline provider/adapters/src/fallback/mod.rs:289]: Nit:

`LocaleFallbacker` is intended to run a specific algorithm (vertical fallback) on a specific type (a locale). It should not know what a `DataRequest` is. The data provider part should be isolated in `LocaleFallbackProvider`.

If we find that we want to pass additional metadata into this function, we can either add them to the `LocaleFallbackConfig` or create another function here that takes the additional metadata as another argument.

Also, if the first line of a function is `.clone()` it almost always means that you should take that thing by value in the function.

All this is to say, please change this function to take `locale: DataLocale`

**Review by sffc — COMMENTED** [review]
Praise: Good improvement! One small issue and otherwise LGTM

**Review by sffc — APPROVED** [review]
Thank you :)

**Pull request merged**